### PR TITLE
Feature: Update link styling

### LINKED ISSUE #3918: Look into changing TOP's link color on lightmode
Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this issue follows the `brief description of request` format, e.g. `Add dark mode to website`

The following checkbox is OPTIONAL:

-   [x] I would like to be assigned this issue to work on it

<hr>

**1. Description of the Feature Request:**
During #3915 I ran Google Lighthouse to test the accessibility impact of those changes, and discovered as a side effect that our current light mode link color (tailwind gold-600) doesn't meet the WCAG AA standards for contrast, and the same goes for it as a background with white text (on eg the "Next Lesson" button). In that PR @KevinMulhern mentioned

> Thanks for reporting that back @Asartea. I think we should start a separate discussion about changing the link colors. There are some UX improvements beyond just improving accessibility we could gain from changing to a different color.

We should look into if we want to change those colors, and if so to what

**2. Acceptance Criteria:**
```[tasklist]
### Acceptance Criteria
- [x] A color is decided upon, which should ideally apply to as much of the light mode as possible
- [ ] All currently inacessible elements are converted to this new color
```

### PR DESCRIPTION
**Pull request by Asartea**
## Because
- Our current link colors don't meet basic contrast requirements

## This PR
- Updates our link styles:
  - Changes link colors to `blue-800` and `blue-300` for light mode and dark mode respectively, and the matching `purple` colors for visited text; all colors now meet WCAG AAA normal against the relevant background
  - Changes the underline style from underline normally and none on hover to none normally and underline on hover; this behavior was inconsistent both within the rest of TOP, and with how most major websites (ie Google, Wikipedia) handle it

## Issue
Closes #3918 

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `keyword: brief description of change` format, using one of the following keywords:
  - `Feature` - adds new or amends existing user-facing behavior
  - `Chore` - changes that have no user-facing value, refactors, dependency bumps, etc
  - `Fix` - bug fixes
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] I have verified all tests and linters pass after making these changes.
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section

### CODE REVIEW

File: app/javascript/stylesheets/lesson-content.css [hunk 1]
```diff
@@ -79,7 +79,11 @@
   }
 
   .lesson-content a[href^="http"] {
-    @apply hover:no-underline;
+    @apply no-underline hover:underline;
```
Comment (KevinMulhern) [inline app/javascript/stylesheets/lesson-content.css:82]: Whats the reason for this change?

**Review by KevinMulhern — COMMENTED** [review]
Colors look perfect, nice work 🔥 

Just a question about switching the hover underline.

File: app/javascript/stylesheets/lesson-content.css (hunk 1 above)
Comment (Asartea) [reply app/javascript/stylesheets/lesson-content.css:82]: Threefold:
- This is how we tackle it everywhere else within the TOP websites; its only the in lesson links that diverge
- The new link colors in my experience tend to feel a bit bolder, and especially in areas where a lot of links where clustered together it occasionally started feeling cramped; by only showing the underline on hover we give everything a bit more space
- It just feels (at least to me) clearer if something gets more prominent on hover than not

Comment (KevinMulhern) [reply app/javascript/stylesheets/lesson-content.css:82]: Thanks for laying it out. I think it was just a little bit jarring for me at first glance. Too used to looking at how it was I suppose lol. But after actually reading through some of the content, yep this is better, good call 💪

**Review by KevinMulhern — APPROVED** [review]

**Comment by an4s911** [comment]
image

Why are some underlined and others not?

**Pull request merged**